autopilot/contractor.go

```diff
@@ -135,10 +135,12 @@ func (c *contractor) performContractMaintenance(ctx context.Context, cfg api.Aut
 	}
 
 	// min score to pass checks.
-	redundancy := rs.TotalShards / rs.MinShards
-	minScore, err := c.managedFindMinAllowedHostScores(ctx, cfg, hosts, storedData, float64(redundancy))
-	if err != nil {
-		return fmt.Errorf("failed to determine min score for contract check: %w", err)
+	var minScore float64
+	if len(hosts) > 0 {
+		minScore, err = c.managedFindMinAllowedHostScores(ctx, cfg, hosts, storedData, rs.Redundancy())
+		if err != nil {
+			return fmt.Errorf("failed to determine min score for contract check: %w", err)
+		}
 	}
```

Comment:
(nit) 0 hosts should hardly ever happen so I think we want at least a warning here if it does.